core/src/test/scala/zio/config/ReadErrorsTest.scala

```diff
@@ -36,4 +41,82 @@ object ReadErrorsTestUtils {
       list <- Gen.listOfN(n)(genReadError)
     } yield list
   }
+
+  val simpleMissingValue    = ReadError.MissingValue(List(Step.Key("k1"), Step.Key("k2"), Step.Index(1), Step.Key("k3")))
+  val simpleFormatError     = ReadError.FormatError(List(Step.Key("k1"), Step.Key("k2")), "Format error")
+  val simpleConversionError = ReadError.ConversionError(List(Step.Key("k1"), Step.Key("k2")), "Conversion error")
+  val complexErrorsForPrettyPrint = ReadError.AndErrors(
+    List(
+      simpleMissingValue,
+      simpleFormatError,
+      simpleConversionError,
+      ReadError.OrErrors(
+        List(
+          ReadError.AndErrors(
+            List(ReadError.OrErrors(List(simpleMissingValue, simpleMissingValue)), simpleFormatError)
+          ),
+          ReadError.OrErrors(List(simpleMissingValue, simpleMissingValue)),
+          simpleConversionError
+        )
+      ),
+      ReadError.AndErrors(List(ReadError.OrErrors(List(simpleMissingValue, simpleMissingValue)), simpleFormatError))
+    )
+  )
+
+  val complexErrorsPrettyPrint =
+    """ReadError failed.
+      |╥
+      |╠══╦══╦══╦══╦══╦══╗
```

Comment:
Is there a limit on widht? There might be a lot more errors.

Reply:
For instance: in ADT you'll get N branches for N subclasses.

Reply:
There is no limit on width. The parallel lines number is same as the total errors in AndErros.